### PR TITLE
Changed scope of CmdWatchSubscriptionChan to local

### PR DESCRIPTION
**Pull request by KanniShashankh**
it's not global anymore, this might be a noobish implemention tho, i'm open for advice on how to implement it in a better way. Thanks!

Fixes https://github.com/DiceDB/dice/issues/1000
CmdWatchSubscriptionChan shouldn't be a global variable

### CODE REVIEW

**Comment by KanniShashankh** [comment]
also, i think my linter might be misconfigured or something, the diff shows extra lines touched. :(

**Comment by KanniShashankh** [comment]
am i missing something here? i'll fix the linter ci run fail but for the integration tests. it fails for even the base branch? is there some config that i'm missing?
Screenshot 2024-10-21 at 11 25 27 PM

**Comment by JyotinderSingh** [comment]
Looks like you may have messed up the merge with master. Usually, you want to:
1. update your local master with the upstream `git checkout master && git pull upstream`
2. checkout your feature branch `git checkout <branch name>
3. merge with master `git merge master`
4. push `git push`

File: integration_tests/commands/resp/setup.go [hunk 1]
```diff
@@ -127,8 +128,11 @@ func RunTestServer(wg *sync.WaitGroup, opt TestServerOptions) {
 	gec := make(chan error)
 	shardManager := shard.NewShardManager(1, queryWatchChan, cmdWatchChan, gec, logr)
 	workerManager := worker.NewWorkerManager(20000, shardManager)
+	var (
+		cmdWatchSubscriptionChan chan watchmanager.WatchSubscription
+	)
```
Comment (JyotinderSingh) [inline integration_tests/commands/resp/setup.go:133]: declare this in the same manner as the queryWatchChan and cmdWatchChan to maintain consistency. Ideally these definitions should be clubbed together.

File: internal/server/httpServer.go
Comment (JyotinderSingh) [inline internal/server/httpServer.go:1]: you can remove these changes as they are unrelated.

File: internal/server/websocketServer.go
Comment (JyotinderSingh) [inline internal/server/websocketServer.go:1]: undo this unrelated change.

File: internal/watchmanager/watch_manager.go [hunk 2]
```diff
@@ -18,30 +18,30 @@ type (
...
 
 	Manager struct {
-		querySubscriptionMap map[string]map[uint32]struct{}              // querySubscriptionMap is a map of Key -> [fingerprint1, fingerprint2, ...]
-		tcpSubscriptionMap   map[uint32]map[chan *cmd.DiceDBCmd]struct{} // tcpSubscriptionMap is a map of fingerprint -> [client1Chan, client2Chan, ...]
-		fingerprintCmdMap    map[uint32]*cmd.DiceDBCmd                   // fingerprintCmdMap is a map of fingerprint -> DiceDBCmd
-		logger               *slog.Logger
+		querySubscriptionMap     map[string]map[uint32]struct{}              // querySubscriptionMap is a map of Key -> [fingerprint1, fingerprint2, ...]
+		tcpSubscriptionMap       map[uint32]map[chan *cmd.DiceDBCmd]struct{} // tcpSubscriptionMap is a map of fingerprint -> [client1Chan, client2Chan, ...]
+		fingerprintCmdMap        map[uint32]*cmd.DiceDBCmd                   // fingerprintCmdMap is a map of fingerprint -> DiceDBCmd
+		cmdWatchSubscriptionChan chan WatchSubscription
```
Comment (JyotinderSingh) [inline internal/watchmanager/watch_manager.go:24]: We can add a comment here explaining the purpose of this channel, same as the other members of this struct.

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for this contribution @KanniShashankh! The changes look good overall, have left a few minor comments.

**Pull request merged**